modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,486 @@
+##' Download NEON Site Met CSV files
+##'
+##' @name download.NEONmet
+##' @title download.NEONmet
+##' @export
+##' @param site the NEON ID of the site to be downloaded, used as file name prefix. 
+##' The 4-letter SITE code  in \href{http://www.neonscience.org/science-design/field-sites/list}{list of NEON sites}
+##' @param outfolder location on disk where outputs will be stored
+##' @param start_date the start date of the data to be downloaded. Format is YYYY-MM-DD (will only use the year and month of the date)
+##' @param end_date the end date of the data to be downloaded. Format is YYYY-MM-DD (will only use the year and month part of the date)
+##' @param overwrite should existing files be overwritten
+##' @param verbose should the function be very verbose
+##' @importFrom PEcAn.utils fqdn logger.debug logger.error logger.warn logger.severe
+
+download.NEONmet <- function(sitename, outfolder, start_date, end_date, 
+                                  overwrite = FALSE, verbose = FALSE,  ...) {
+
+  if (!file.exists(outfolder)) {
+    dir.create(outfolder)
+  }
+  
+  #Check if site is a NEON site
+  site <- sub(".* \\((.*)\\)", "\\1", sitename)
+  siteinfo <- nneo::nneo_site(site)
+  if (!exists("siteinfo")) {
+    logger.error("Could not get information about", sitename, ".", "Is this a NEON site?")
+  }
+  
+  #See what products and dates are available for this site
+  availProducts <- siteinfo$dataProducts$dataProductCode #list of data prodcuts by code
+  availDates <- siteinfo$dataProducts$availableMonths #lists of availably YYYY-MM per site, use unlist()
+  lat <- siteinfo$siteLatitude
+  lon <- siteinfo$siteLongitude
+  
+  #Figure out which months are needed
+  start_date <- as.POSIXlt(start_date, tz = "UTC")
+  end_date <- as.POSIXlt(end_date, tz = "UTC")
+  start_ymd <- lubridate::ymd(start_date)
+  end_ymd <- lubridate::ymd(end_date)
+
+  #Subset to max/min available dates if start or end date exceed those bounds
+  allDates = unlist(availDates)
+  minDate <- min(allDates)
+  maxDate <- max(allDates)
+  start_ym = substr(start_ymd,1,7)
+  end_ym = substr(end_ymd,1,7)
```

Comment:
I really wish RStudio would just automatically change one into the other when typed!